### PR TITLE
Deprecated Readonly resources on Puppet + allow their references to accept string

### PR DESCRIPTION
**Pull request by rambleraptor**
We decided that `readonly` resources like gcompute_zone, gcompute_region aren't strictly necessary.

Step 1: Allow ResourceRefs referencing `readonly` resources to accept strings (this PR)
Step 2: Add deprecation messages to `readonly` resources (this PR)
Step 3: In a future release, change `deprecated: true` to `exclude: true`

-----------------------------------------------------------------
# [all]
Deprecated Readonly resources on Puppet + allow their references to accept string
## [puppet]
Deprecated Readonly resources + allow their references to accept string

### CODE REVIEW

File: products/spanner/puppet.yaml [hunk 1]
```diff
@@ -78,28 +78,9 @@ overrides: !ruby/object:Provider::ResourceOverrides
         )
     provider_helpers:
       - 'products/spanner/helpers/ruby/instance_helpers.rb.erb'
+  # Readonly resources.
   InstanceConfig: !ruby/object:Provider::Puppet::ResourceOverride
-    handlers: !ruby/object:Provider::Puppet::Handlers
```
Comment (ndmckinley) [inline products/spanner/puppet.yaml]: This still breaks existing configs, right?

Comment (rambleraptor) [reply products/spanner/puppet.yaml]: Oh, shoot. It does. One second.

Comment (rambleraptor) [reply products/spanner/puppet.yaml]: Fix made! No code changes should occur other than adding the deprecation message for those particular providers.

**Comment by ndmckinley** [comment]
Can you rebase and rerun Travis?  There's a fix for that issue merged into master.

File: templates/puppet/provider.erb [hunk 2]
```diff
@@ -60,6 +60,13 @@ Puppet::Type.type(:<%= object.out_name -%>).provide(:google) do
   end
 
   def self.prefetch(resources)
+<% if object.deprecated -%>
+    Puppet.warning [
+      "<%= object.out_name -%> will be deprecated in a future release.",
```
Comment (ndmckinley) [inline templates/puppet/provider.erb]: This message is mostly fine, but I don't think users will know whether they need full URLs or just partial ones, or even just names.  Can you specify?

**Pull request merged**